### DISCUSSION TITLE
Best ways of looking at metabolic flux across cohorts

### DISCUSSION THREAD

**Discussion by manas-kohli** [SysBioChalmers/Human-GEM#531] [Q&A] [answered]
Hey there, I just wanted to ask a few questions regarding looking at FBA for data across cohorts. 

Recently, I created quite a few context-specific models using various breast cancer cohorts including the complete TCGA-BRCA cohort and a few other cohorts mixed in. After creating these and their enzyme constrained counterparts, I performed FBA optimising for growth as is typically done. Now typically, from what I've seen in the literature, flux values obtained this way are used in a comparison way (e.g. comparing cancer with normal, a mutant vs wild-type etc.) but not too many people have used the flux values themselves as a fingerprint for looking at metabolic activity across cohorts. So. for example, I naively applied PCA to flux values obtained from FBA solutions to see if I can get something obvious like metabolic clusters. I do realise that FBA optimising growth represents a really small portion of the flux landscape for each model but the other ways of looking at flux through randomsampling and FVA are not feasible time wise to run for entire cohorts (about 1500+ samples). When I first did this, I obtained two nice clusters but upon looking at my analysis more closely, I realised that I did not scale the flux values for each reaction such that the fluxes at extreme ranges i.e. -1000 and 1000 were the predominant ones causing the variation. I wasn't sure whether I should scale the flux values so that each feature i.e. reaction gets equal importance. When I do that, the landscape looks a little messy and unclear. I then decided to look at fluxes of enzyme constrained models because I thought they may be more meaningful flux values. When I scale enzyme-constraint fluxes from FBA, the PCA looks a lot nicer but I'm not sure if what I'm doing is appropriate.

So from the little bit of a ramble, here are my actual questions:

1) Can you use FBA derived fluxes to look at metabolism in an unsupervised way at all or is this a foolish endeavour and the best way of looking at flux is to compare models to each other?

If the answer to 1) is yes it's not a good idea:

2) Is there an appropriate way through FBA or related techniques to look at flux in an unsupervised manner?

If the answer to 1) is that it's perfectly fine to look at flux through FBA or something similar:

3) When I compare fluxes for models, should I ignore reactions that have unconstrained flux i.e. -1000 or 1000 given they probably represent some futile cycle or something like that?
4) Are ecModels better models to use for this rather than flux?
5) Should I scale my flux values to give equal importance to each reaction? I suspect the answer to this may depend on 3).

Any help would be really appreciated! I've been grappling with this problem for some time so any advice can really benefit me here

**Comment by JonathanRob** [comment, marked as answer]
Hi @manas-kohli, some great questions! In general, I think you are very much on the right track.

> 1. Can you use FBA derived fluxes to look at metabolism in an unsupervised way at all or is this a foolish endeavour and the best way of looking at flux is to compare models to each other?

Yes, you can and I think it is potentially more interesting/meaningful in some cases to do so. The challenges/questions you encountered as you attempted to use the fluxes are often a reason why model structures are compared instead, since it's generally more straightforward and involves fewer assumptions. But that doesn't mean it is incorrect to do so!

> 3. When I compare fluxes for models, should I ignore reactions that have unconstrained flux i.e. -1000 or 1000 given they probably represent some futile cycle or something like that?

If you aren't using parsimonious FBA (pFBA) or some option in your FBA to help reduce the magnitude of fluxes when possible, then I would recommend doing so (e.g., `solveLP(model, 1)` in RAVEN)). This should eliminate extreme fluxes caused by (some) cycles. Otherwise, there is not really a correct way to deal with these unconstrained fluxes. I generally would avoid simply ignoring them, since chances are that they are quite important to the solution/network. One approach is to try and more tightly constrain the exchange reactions to limit metabolite uptake to only a defined set of metabolites (e.g., those in Ham's media as defined in the metabolic task list), and even limit the flux of their uptake (e.g., do not allow glucose uptake to exceed 10 mmol/gdw/h). Unfortunately the decisions made on which constraints to apply will heavily impact the resulting flux distributions and the comparison between your models, so be sure that the constraints are meaningful or justified in some way.

> 4. Are ecModels better models to use for this rather than flux?

I would recommend using ecModels, since they help to apply constraints as I suggested above, but in a more systematic and biologically meaningful manner. So it gives both the advantage of reducing or eliminating unbounded fluxes in the FBA solution, while also yielding flux values that are more biologically meaningful.

> 5. Should I scale my flux values to give equal importance to each reaction? I suspect the answer to this may depend on 3).

I'm not sure I completely understand, but if you are asking if you should scale all reaction fluxes to 0 or +/-1 for example, then the answer is you could. This would then yield a comparison of the structure of the active flux network involved in the chosen objective (biomass production in your example), which may be informative. Though I suspect that in doing so you will lose a lot of resolution and the model differences may be driven by less meaningful reactions that are somewhat "noisy" or random in whether they are used in an FBA solution. So yes, you could, but I would recommend doing it _in addition to_ an analysis of the unscaled fluxes.


But it's definitely an interesting (and challenging) approach worth pursuing, so I encourage you to try different methods and even search the literature to see what others have tried for the same or similar issues.

**Comment by manas-kohli** [comment]
Dear Jonathan,

Thank you for the response, this is really helpful. I'll try the different approaches as you mentioned and see what comes out of the analysis
